### PR TITLE
Asset wrapper and fee updates

### PR DESCRIPTION
**Pull request by tolak**
- Add struct AssetProperties in AssetRegistryInfo
- Set metadata when register assets
- Return proper fee amount when decimals of asset if not 12

### CODE REVIEW

File: scripts/js/xtransfer_index.js [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+    return {
+        x2History: x2History,
+        x3History: x3History
+    };
+}
+
+function karuraHistoryFilter(api, karuraSendHistory) {
+    let x2History = [];
+    let x3History = [];
+    karuraSendHistory.nodes.filter(history => {
```
Comment (h4x3rotab) [inline scripts/js/xtransfer_index.js:265]:
```suggestion
    karuraSendHistory.nodes.forEach(history => {
```

File: scripts/js/xtransfer_index.js [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+        );
+    }
+  
+    return data;
+}
+
+function evmHistoryFilter(api, evmHistory) {
+    let x2History = [];
+    let x3History = [];
+    evmHistory.bridgeOutboundingRecords.filter(history => {
```
Comment (h4x3rotab) [inline scripts/js/xtransfer_index.js:230]:
```suggestion
    evmHistory.bridgeOutboundingRecords.forEach(history => {
```

File: pallets/xtransfer/src/bridge_transfer/mod.rs [hunk 3]
```diff
@@ -516,25 +516,45 @@ pub mod pallet {
...
+									))
+								} else {
+									Some(fee_in_decimal_12.saturating_div(
+										10u128.saturating_pow(12 - decimals as u32),
+									))
+								};
+							} else {
+								fee_in_asset = None
+							}
+							fee_in_asset
```
Comment (h4x3rotab) [inline pallets/xtransfer/src/bridge_transfer/mod.rs:556]: Can we make it a separate function and add a few unit tests? Also, I suggest to use `fee_e12` or `fee_e18` to simplify the variable names

Comment (tolak) [reply pallets/xtransfer/src/bridge_transfer/mod.rs]: Done

File: scripts/js/xtransfer_index.js (hunk 2 above)
Comment (tolak) [reply scripts/js/xtransfer_index.js:230]: Done

File: scripts/js/xtransfer_index.js (hunk 1 above)
Comment (tolak) [reply scripts/js/xtransfer_index.js:265]: Done

File: pallets/xtransfer/src/bridge_transfer/mod.rs [hunk 4]
```diff
@@ -516,25 +516,45 @@ pub mod pallet {
...
 			match (&asset.id, &asset.fun) {
-				(Concrete(asset_id), Fungible(amount)) => {
-					let fee_estimated_in_pha =
-						Self::estimate_fee_in_pha(chain_id, (*amount).into());
-					if T::NativeChecker::is_native_asset(asset) {
-						Some(fee_estimated_in_pha.into())
+				(Concrete(location), Fungible(amount)) => {
+					let id = T::AssetsWrapper::id(&XTransferAsset(location.clone()));
+					if id.is_none() {
+						None
```
[the remark this answers is not in the document]
Comment (tolak) [reply pallets/xtransfer/src/bridge_transfer/mod.rs]: Done

File: pallets/xtransfer/src/assets_wrapper.rs [hunk 5]
```diff
@@ -399,5 +451,9 @@ pub mod pallet {
 		fn lookup_by_resource_id(resource_id: &[u8; 32]) -> Option<XTransferAsset> {
 			AssetByResourceIds::<T>::get(resource_id)
 		}
+
+		fn decimals(id: &<T as pallet_assets::Config>::AssetId) -> Option<u8> {
+			RegistryInfoByIds::<T>::get(&id).map_or(None, |m| Some(m.properties.decimals))
```
[the remark this answers is not in the document]
Comment (tolak) [reply pallets/xtransfer/src/assets_wrapper.rs]: Done

**Review by h4x3rotab — APPROVED** [review]
Please simplify fee calculation

**Review by h4x3rotab — APPROVED** [review]
LGTM

**Pull request merged**